File: nicegui/elements/file_picker.js
export default {
    template: `
        <q-file
            v-model="file"
            @update:modelValue="file_pick"

            use-chips

            :label="label"
            :multiple="multiple"
            :accept="accept"
        />
    `,

    props: {
        label: String,
        multiple: Boolean,
        accept: String,
    },

    setup() {
        return {
            file: Vue.ref(null),
        };
    },

    methods: {
        file_pick() {
            setTimeout(() => {
                const m_file = this._.subTree.props.modelValue;

                if (!m_file) return;
                if (this.file_url) URL.revokeObjectURL(this.file_url);

                this.file_url = URL.createObjectURL(m_file);

                const m_return = {
                    name: m_file.name,
                    size: m_file.size,
                    type: m_file.type,
                    url: this.file_url,
                    last_modified: m_file.lastModified,
                };

                this.$emit("file_pick", m_return);
            }, 0);
        },
    },
};

File: nicegui/elements/file_picker.py
from ..dependencies import register_component
from ..element import Element

register_component("file_picker", __file__, "file_picker.js")


class FilePicker(Element):
    def __init__(self, label="File...", multiple=False, accept_types=[""]) -> None:
        """File Picker

        A component which handles the user interaction for picking file

        :param label: A text label that will “float” up above the input field, once the field gets focus
        :param multiple: Whether or not to allow multiple files to be selected
        :param accept_types: A list of file types to accept

        API to get information from the selected file:

        :event file_pick: Triggered when the user selects a file
        :return: List of dicitionaries with the following keys: name, size, type, url, last_modified
        :save: This list is saved in file property of the component
        """
        super().__init__("file_picker")

        self._props["label"] = label
        self._props["multiple"] = multiple
        self._props["accept"] = ", ".join(accept_types)

        self.file = {"name": "", "size": "", "type": "", "url": "", "last_modified": ""}
        self.on("file_pick", self.handle_file_pick)

    def handle_file_pick(self, data) -> None:
        self.file = data["args"]
